example-app/app/Actions/CreateFlightAction.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Actions;
+
+use App\Http\Requests\StoreFlightRequest;
+use App\Models\Flight;
+
+class CreateFlightAction
+{
+    public function __invoke(StoreFlightRequest $request): Flight
```

Comment:
Do not pass the request inside you app logic. As a general rule, the request must live only on the controller, it should not be passed to other layers on you app.

You can create a DTO, at the same level of the Actions/ folder, you can create a DataTransferObjets/ folder and create a CreateFlightDto class

Reply:
Also, do not use the __invoke() for actions, create a method called execute() for put the logic

This is part of our tech decisions
https://lightit.slite.com/app/docs/dpqMFsErCJrCd_